perllib/FixMyStreet/Script/ArchiveOldEnquiries.pm

```diff
@@ -141,6 +141,11 @@ sub close_problems {
     my $extra = { auto_closed_by_script => 1 };
     $extra->{is_superuser} = 1 if !$opts->{user_name};
 
+    # need to do this again in case no reports were closed with an
+    # email in which case we won't have set the lang and domain
+    my $cobrand = FixMyStreet::Cobrand->get_class_for_moniker($opts->{cobrand})->new();
+    $cobrand->set_lang_and_domain($problems->first->lang, 1);
```

Comment:
This will fail in the situation that `$problems->first` doesn't exist. Better to reuse the `my @problems = $problems->all; return if scalar(@problems) == 0;` pattern from `send_email_and_close` above?

Reply:
(or more simply `return unless $problems->count;` perhaps?)

Reply:
Ah yes, good point.